cupy/cuda/compiler.py

```diff
@@ -1,36 +1,34 @@
 import hashlib
 import os
 import re
-import subprocess
-import sys
 import tempfile
 
 import filelock
 import six
 
 from cupy.cuda import device
 from cupy.cuda import function
+from pynvrtc.compiler import NVRTCInterface
+from pynvrtc.compiler import Program
 
+_nvrtc_version = None
 
-_nvcc_version = None
 
+def _get_nvrtc_version():
+    global _nvrtc_version
+    if _nvrtc_version is None:
+        interface = NVRTCInterface()
+        _nvrtc_version = interface.nvrtcVersion()
 
-def _get_nvcc_version():
-    global _nvcc_version
-    if _nvcc_version is None:
-        cmd = ['nvcc', '--version']
-        _nvcc_version = _run_nvcc(cmd, '.')
-
-    return _nvcc_version
+    return _nvrtc_version
 
 
 def _get_arch():
```

Comment:
compute_xx is used to specify a virtual GPU architecture, while sm_xx is used to specify the actual real architecture for which a given code/PTX will be compiled. NVRTC generates PTX file for virtual architecture, and then the CUDA driver converts the PTX file for each real GPU architecture by generating an optimized binary.